src/main/java/hu/oe/nik/szfmv/automatedcar/model/deserializer/Deserializer.java

```diff
@@ -0,0 +1,129 @@
+package hu.oe.nik.szfmv.automatedcar.model.deserializer;
+
+import com.google.gson.*;
+import com.google.gson.reflect.TypeToken;
+import hu.oe.nik.szfmv.automatedcar.model.ReferencePoint;
+import hu.oe.nik.szfmv.automatedcar.model.WorldObject;
+
+import java.io.*;
+import java.lang.reflect.Type;
+import java.util.*;
+
+public class Deserializer {
+    private class WorldObjectHelper {
+        public String type;
+        public int x;
+        public int y;
+        public float m11;
+        public float m12;
+        public float m21;
+        public float m22;
+
+        public WorldObject ParseToWorldObject() {
+            var object = new WorldObject();
+            object.setType(type);
+            object.setX(x);
+            object.setY(y);
+            var rotMatrix = new float[2][2];
+            rotMatrix[0][0] = m11;
+            rotMatrix[0][1] = m12;
+            rotMatrix[1][0] = m21;
+            rotMatrix[1][1] = m22;
+            object.setRotationMatrix(rotMatrix);
+            object.initObject();
+
+            return object;
+        }
+    }
+
+    private static void TestFile(String fileName) throws IllegalArgumentException {
+        if (!fileName.contains("json")) {
+            throw new IllegalArgumentException("Provided file's extension is not .json! \n File:" + fileName);
+        }
+    }
+
+    private static String ReadFile(String fileName) {
+        var content = new StringBuilder();
+        try {
+            var reader = new BufferedReader(new FileReader(ClassLoader.getSystemResource(fileName).getFile()));
+            String line;
+            while ((line = reader.readLine()) != null) {
+                content.append(line);
+            }
+
+            return content.toString();
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    private static WorldObject AddLayeringWithStaticInfo(WorldObject object) {
+        if ((object.getType().contains("road") && !object.getType().contains("roadsign"))
+                || object.getType().contains("garage")
+                || (object.getType().contains("parking") && !object.getType().contains("roadsign"))) {
+            object.setIsStatic(true);
+            object.setZ(0);
+        }
+
+        if (object.getType().contains("man")
```

Comment:
Ez is elég randa. Ezt Pythonban így írnám:
```python
if object.getType() in ["man", "car", "bicycle", ...]:
    # do something
```
Most nem fogom kikeresni a Java megfelelőjét, de miért nem lehet azt megnézni, hogy a getType() eredménye benne van-e egy listában ahelyett, hogy 10-szer lemásolod ugyanazt?

**REFACTOR lépés hiányzik!!!**

Reply:
Ha a `.getType()` `string` típusú és pontos egyenlőséget akarunk nézni (`String.equals(String)`), akkor én `string` `switch`-re cserélném _(amit meg külön metódusként ki-extract-olnék majd)_:
```java
switch (object.getType()) {
    case "man":
    case "car":
    case "bicycle":
    case "bollard":
    case "boundary":
    case "tree":
    case "roadsign":
        object.setZ(1);
        if (!object.getType().equals("man")) {
            object.setIsStatic(true);
        }
        break;
}
```

Egyébként meg lehet szebb lenne egy `enum`-ba tenni a típusokat:
```java
enum ObjectType {
    MAN, CAR, BICYCLE, BOLLARD, BOUNDARY, TREE, ROAD_SIGN
}
```
Ha valami fájlból van beolvasva ez a típus, akkor egy helyen map-elném, pl.:
```java
enum ObjectType {
    MAN, CAR, BICYCLE, BOLLARD, BOUNDARY, TREE, ROAD_SIGN;
    public static ObjecType parseObjectType(String strToParse) {
        //....
    }
}
```
Aztán betöltéskor valahogy így lenne használva:
```java
import static containing.package.path.ObjectType.parseObjectType;
//...

//...
String typeRead = ...; //json-ből vagy xml-ből olvasás v. olyasmi
//...
object.setType(parseObjecType(typeRead))
//...
```
Ezek csak ötletek, hátha hasznosnak találjátok.

Reply:
Mindegyik sokkal szebb mint ami ott van.